### PR TITLE
Support to pull prompt templates and model configuration from MakerSuiteHub

### PR DESCRIPTION
**Pull request by afirstenberg**
Addresses #2564 

This creates a MakerSuiteHub class with methods to load (with cache) a text, "data", or chat prompt that has been saved via makersuite.google.com. This prompt has methods to convert to a template, a prompt-free model, or to a chain which combines the two.

There are also some changes in the base connection class and types to allow for more generic calls to other Google REST APIs (in this case, Google Drive).

Examples and other documentation are pending initial review to make sure this looks good, but the tests should be complete.

### CODE REVIEW

File: langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    const result = await connection.request();
+    const ret = new MakerSuitePrompt(result.data);
+    this.cache[id] = {
+      prompt: ret,
+      updated: Date.now(),
+    };
+    return ret;
+  }
+
+  async load(id: string): Promise<MakerSuitePrompt> {
```
Comment (jacoblee93) [inline langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts:353]: Strong preference between `load` vs. `pull`?

File: langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+            content: exchange.response,
+          };
+        }
+        return example;
+      })
+      .filter((value) => Object.keys(value).length);
+    return ret;
+  }
+
+  toModel(): LLM | BaseChatModel {
```
Comment (jacoblee93) [inline langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts:235]: This can just be `BaseLanguageModel`

**Comment by jacoblee93** [comment]
So sorry about the delay - this seems reasonable, just missing entrypoints and docs

File: langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts [hunk 3]
```diff
@@ -0,0 +1,358 @@
+import { protos } from "@google-ai/generativelanguage";
```
Comment (jacoblee93) [inline langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts:1]: Use `import type`

File: langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+import { google } from "@google-ai/generativelanguage/build/protos/protos.js";
```
Comment (jacoblee93) [inline langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts:2]: Here as well I think?

File: langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts (hunk 1 above)
Comment (afirstenberg) [reply langchain/src/experimental/hubs/makersuite/googlemakersuitehub.ts:353]: Changing to pull() and forcePull()

**Comment by afirstenberg** [comment]
@jacoblee93 - I think (hope) this is ready for final review and merge.

**Comment by jacoblee93** [comment]
Thank you, will look tomorrow!

**Comment by jacoblee93** [comment]
Should be good to go! Thank you

**Pull request merged**